pkg/api/metrics.go

```diff
@@ -230,6 +230,11 @@ func (hs *HTTPServer) createRequest(ctx context.Context, ds *models.DataSource,
 		if token := hs.OAuthTokenService.GetCurrentOAuthToken(ctx, query.User); token != nil {
 			delete(query.Headers, "Authorization")
 			query.Headers["Authorization"] = fmt.Sprintf("%s %s", token.Type(), token.AccessToken)
+
+			idToken := token.Extra("id_token")
+			if idToken != nil {
+				query.Headers["X-ID-Token"] = idToken.(string)
```

Comment:
Are we sure the ID token is always a string? Otherwise something like this might be preferable.

```suggestion
			idToken, ok := token.Extra("id_token").(string)
			if ok && idToken != "" {
				query.Headers["X-ID-Token"] = idToken
```

Reply:
Sounds reasonable